astropy/io/fits/header.py

```diff
@@ -350,13 +350,29 @@ def fromstring(cls, data, sep=''):
 
         cards = []
 
+        if not isinstance(sep, str):
+            sep = sep.decode('latin1')
+
         # If the card separator contains characters that may validly appear in
         # a card, the only way to unambiguously distinguish between cards is to
         # require that they be Card.length long.  However, if the separator
         # contains non-valid characters (namely \n) the cards may be split
         # immediately at the separator
         require_full_cardlength = set(sep).issubset(VALID_HEADER_CHARS)
 
+        if isinstance(data, bytes):
```

Comment:
I'd add a comment here too, like in `card.py`.

Reply:
👍

Reply:
Done.